mobile/src/navigations/root/root.tsx

```diff
@@ -21,7 +22,14 @@ const Root: React.FC = () => {
     <NativeStack.Navigator screenOptions={screenOptions}>
       <NativeStack.Screen name={RootScreenName.SIGN_IN} component={Auth} />
       <NativeStack.Screen name={RootScreenName.SIGN_UP} component={Auth} />
-      <NativeStack.Screen name={RootScreenName.MAIN} component={Main} />
+      <NativeStack.Screen
+        name={RootScreenName.MAIN}
+        component={(): JSX.Element => (
+          <ProtectedRoute>
```

Comment:
Let's leave this component but don't wrap anything for now, we will use it a bit later